File: FFRv2/app/src/main/java/com/devingotaswitch/rankings/domain/Team.java
package com.devingotaswitch.rankings.domain;

import com.amazonaws.util.StringUtils;
import com.devingotaswitch.utils.Constants;

public class Team {

    private String name;
    private String oLineRanks;
    private String draftClass;
    private int qbSos;
    private int rbSos;
    private int wrSos;
    private int teSos;
    private int dstSos;
    private int kSos;
    private String bye;
    private String incomingFA;
    private String outgoingFA;
    private String schedule;

    private static final String FA_DELIMITER = "~~~";

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getoLineRanks() {
        return oLineRanks;
    }

    public void setoLineRanks(String oLineRanks) {
        this.oLineRanks = oLineRanks;
    }

    public String getDraftClass() {
        return draftClass;
    }

    public int getQbSos() {
        return qbSos;
    }

    public void setQbSos(int qbSos) {
        this.qbSos = qbSos;
    }

    public int getRbSos() {
        return rbSos;
    }

    public void setRbSos(int rbSos) {
        this.rbSos = rbSos;
    }

    public int getWrSos() {
        return wrSos;
    }

    public void setWrSos(int wrSos) {
        this.wrSos = wrSos;
    }

    public int getTeSos() {
        return teSos;
    }

    public void setTeSos(int teSos) {
        this.teSos = teSos;
    }

    public int getDstSos() {
        return dstSos;
    }

    public void setDstSos(int dstSos) {
        this.dstSos = dstSos;
    }

    public int getkSos() {
        return kSos;
    }

    public void setkSos(int kSos) {
        this.kSos = kSos;
    }

    public void setDraftClass(String draftClass) {
        this.draftClass = draftClass;
    }

    public String getBye() {
        return bye;
    }

    public void setBye(String bye) {
        this.bye = bye;
    }

    public String getOutgoingFA() {
        return outgoingFA;
    }

    public void setOutgoingFA(String outgoingFA) {
        this.outgoingFA = outgoingFA;
    }

    public String getIncomingFA() {
        return incomingFA;
    }

    public void setIncomingFA(String incomingFA) {
        this.incomingFA = incomingFA;
    }

    public String getFaClass() {
        if (!StringUtils.isBlank(incomingFA) && !StringUtils.isBlank(outgoingFA)) {
            return new StringBuilder(incomingFA)
                    .append(FA_DELIMITER)
                    .append(outgoingFA)
                    .toString();
        }
        return "";
    }

    public void setFaClass(String faClass) {
        if (faClass != null) {
            String[] faArr = faClass.split(FA_DELIMITER);
            this.incomingFA = faArr[0];
            this.outgoingFA = faArr[1];
        }
    }

    public String getSchedule() {
        return schedule;
    }

    public void setSchedule(String schedule) {
        this.schedule = schedule;
    }

    public int getSosForPosition(String position) {
        switch (position) {
            case Constants.QB:
                return getQbSos();
            case Constants.RB:
                return getRbSos();
            case Constants.WR:
                return getWrSos();
            case Constants.TE:
                return getTeSos();
            case Constants.DST:
                return getDstSos();
            case Constants.K:
                return getkSos();
        }
        return -1;
    }
}

File: FFRv2/app/src/main/java/com/devingotaswitch/rankings/domain/Team.kt
package com.devingotaswitch.rankings.domain

import com.amazonaws.util.StringUtils
import com.devingotaswitch.utils.Constants

class Team {
    var name: String = ""
    var oLineRanks: String? = null
    var draftClass: String? = null
    var qbSos = 0.0
    var rbSos = 0.0
    var wrSos = 0.0
    var teSos = 0.0
    var dstSos = 0.0
    var kSos = 0.0
    var bye: String? = null
    var incomingFA: String? = null
    var outgoingFA: String? = null
    var schedule: String? = null

    var faClass: String?
        get() = if (!StringUtils.isBlank(incomingFA) && !StringUtils.isBlank(outgoingFA)) {
            incomingFA +
                    FA_DELIMITER +
                    outgoingFA
        } else ""
        set(faClass) {
            if (faClass != null && faClass.contains(FA_DELIMITER)) {
                val faArr = faClass.split(FA_DELIMITER)
                incomingFA = faArr[0]
                outgoingFA = faArr[1]
            }
        }

    fun getSosForPosition(position: String?): Double {
        when (position) {
            Constants.QB -> return qbSos
            Constants.RB -> return rbSos
            Constants.WR -> return wrSos
            Constants.TE -> return teSos
            Constants.DST -> return dstSos
            Constants.K -> return kSos
        }
        return Constants.DEFAULT_SOS
    }

    companion object {
        private const val FA_DELIMITER = "~~~"
    }
}